test/apps/CMakeLists.txt

```diff
@@ -0,0 +1,112 @@
+
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin/test)
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}
+    $<TARGET_PROPERTY:teca_alg,INTERFACE_INCLUDE_DIRECTORIES>
+    $<TARGET_PROPERTY:teca_io,INTERFACE_INCLUDE_DIRECTORIES>
+    )
+
+teca_add_test(test_deeplabv3p_ar_detect_app
+    LIBS teca_io teca_alg
+    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/test_deeplabv3p_ar_detect_app.sh
+    ${CMAKE_BINARY_DIR}/${BIN_PREFIX}
+    --pytorch_deeplab_model ${TECA_DATA_ROOT}/cascade_deeplab_IVT.pt
+    --input_regex "${TECA_DATA_ROOT}/ARTMIP_MERRA_2D.*\.nc$"
+    --filename_time_template "ARTMIP_MERRA_2D_%Y%m%d_%H.nc"
+    --t_axis_variable ''
+    --t_units "days since 1979-01-01 00:00:00"
+    --output_file "${TECA_DATA_ROOT}/test_deeplabv3p_ar_detect_app_output.nc"
+    0 "${TECA_DATA_ROOT}/test_deeplabv3p_ar_detect_app_ref.bin"
+    FEATURES ${TECA_HAS_NETCDF}
+    REQ_TECA_DATA)
+
+teca_add_test(test_bayesian_ar_detect_app
+    LIBS teca_io teca_alg
+    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/test_bayesian_ar_detect_app.sh
+    ${CMAKE_BINARY_DIR}/${BIN_PREFIX}
+    --input_regex "${TECA_DATA_ROOT}/ARTMIP_MERRA_2D.*\.nc$"
+    --cf_reader::filename_time_template "ARTMIP_MERRA_2D_%Y%m%d_%H.nc"
+    --cf_reader::t_axis_variable ''
+    --cf_reader::t_units "days since 1979-01-01 00:00:00"
+    --output_file "${TECA_DATA_ROOT}/test_bayesian_ar_detect_app_output.nc"
+    0 "${TECA_DATA_ROOT}/test_bayesian_ar_detect_app_ref.bin"
+    FEATURES ${TECA_HAS_NETCDF}
+    REQ_TECA_DATA)
+
+teca_add_test(test_tc_detect_app
+    LIBS teca_io teca_alg
+    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/test_tc_detect_app.sh
+    ${CMAKE_BINARY_DIR}/${BIN_PREFIX}
+    --input_regex "${TECA_DATA_ROOT}/test_tc_candidates_1990_07_0[12]\\.nc"
+    --candidate_file "${TECA_DATA_ROOT}/test_tc_candidates_app_output.bin"
+    --track_file "${TECA_DATA_ROOT}/test_tc_track_app_output.bin"
+    --lowest_lat -20 --highest_lat 20
+    "${TECA_DATA_ROOT}/test_tc_candidates_app_ref.bin"
+    "${TECA_DATA_ROOT}/test_tc_track_app_ref.bin" "storm_id"
+    REQ_TECA_DATA)
```

Comment:
this requires netcdf

Reply:
Do you mean as an input? It is set on line 40 `--input_regex "${TECA_DATA_ROOT}/test_tc_candidates_1990_07_0[12]\\.nc"`

The .bin files mentioned on 44 & 45 are reference files to be diffed with app output

Reply:
this test can't run without netcdf, it needs `FEATURES ${TECA_HAS_NETCDF}`

Reply:
oh haha my bad